core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountActorSpec.scala

```diff
@@ -0,0 +1,467 @@
+          Future.successful(false)
+
+        override def setBillingAccountName(googleProjectId: GoogleProjectId, billingAccountName: RawlsBillingAccountName, span: OpenCensusSpan): Future[ProjectBillingInfo] =
+          Future.failed(new RawlsException("You do not have access to this billing account or it does not exist."))
+      }
+
+      WorkspaceBillingAccountActor(dataSource, gcsDAO = gcsDao)
+        .updateBillingAccounts
+        .unsafeRunSync()
+
+      runAndWait {
+        for {
+          lastChange <- BillingAccountChanges.getLastChange(testData.billingProject.projectName)
+          billingProject <- rawlsBillingProjectQuery.load(testData.billingProject.projectName)
+          workspace <- workspaceQuery.findByIdOrFail(testData.v1Workspace.workspaceId)
+        } yield {
+          lastChange.value.googleSyncTime shouldBe defined
+          lastChange.value.outcome.value.isFailure shouldBe true
+
+          billingProject.value.invalidBillingAccount shouldBe true
+          billingProject.value.message shouldBe defined
+
+          workspace.billingAccountErrorMessage shouldBe defined
+          workspace.currentBillingAccountOnGoogleProject shouldBe testData.v1Workspace.currentBillingAccountOnGoogleProject
+        }
+      }
+    }
+
+  it should "not update billing project validity when updating workspaces fails" in
+    withEmptyTestDatabase { dataSource: SlickDataSource =>
+      runAndWait {
+        for {
+          _ <- rawlsBillingProjectQuery.create(testData.billingProject)
+          _ <- workspaceQuery.createOrUpdate(testData.workspace)
+          _ <- rawlsBillingProjectQuery.updateBillingAccount(
+            testData.billingProject.projectName,
+            billingAccount = RawlsBillingAccountName(UUID.randomUUID.toString).some,
+            testData.userOwner.userSubjectId
+          )
+        } yield ()
+      }
+
+      val gcsDao = new MockGoogleServicesDAO("test") {
+        override def rawlsCreatedGoogleProjectExists(projectId: GoogleProjectId): Future[Boolean] =
+          Future.successful(false)
+
+        override def setBillingAccountName(googleProjectId: GoogleProjectId, billingAccountName: RawlsBillingAccountName, span: OpenCensusSpan): Future[ProjectBillingInfo] =
+          Future.failed(new RawlsException("he's dead jim."))
+      }
+
+      WorkspaceBillingAccountActor(dataSource, gcsDAO = gcsDao)
+        .updateBillingAccounts
+        .unsafeRunSync()
+
+      runAndWait {
+        for {
+          lastChange <- BillingAccountChanges.getLastChange(testData.billingProject.projectName)
+          billingProject <- rawlsBillingProjectQuery.load(testData.billingProject.projectName)
+          workspace <- workspaceQuery.findByIdOrFail(testData.workspace.workspaceId)
+        } yield {
+          lastChange.value.googleSyncTime shouldBe defined
+          lastChange.value.outcome.value.isSuccess shouldBe true
```

Comment:
Yeah you're right. I've updated the code so that it combines all errors and writes that as the `MESSAGE` field of the database. Now if an update fails, the failure message will reflect all the reasons why it failed.